test/functional/plugin/lsp_spec.lua

```diff
@@ -2443,6 +2443,90 @@ describe('LSP', function()
       eq(0, lines)
       os.remove(new)
     end)
+    it('Ensures if the new buffer remains unlisted if the old was not in window before', function()
+      local old = tmpname()
+      write_file(old, 'Test content')
+      local new = tmpname()
+      os.remove(new) -- only reserve the name, file must not exist for the test scenario
+      local buflisted = exec_lua(
+        [[
+        local old = select(1, ...)
+        local oldbufnr = vim.fn.bufadd(old)
+        local newbufnr = vim.fn.bufadd(new)
+        local new = select(2, ...)
+        vim.lsp.util.rename(old, new)
+        return vim.bo[newbufnr].buflisted
+      ]],
+        old,
+        new
+      )
+      eq(false, buflisted)
+      os.remove(new)
+    end)
+    it('Ensures if the new buffer remains unloaded if the old was not in window before', function()
```

Comment:
I think we can merge this into the previous case. "new buffer remains unloaded and unlisted if the old was not shown in a window"